pkgs/development/tools/parsing/tree-sitter/default.nix

```diff
@@ -58,32 +38,34 @@ let
 
 in rustPlatform.buildRustPackage {
   pname = "tree-sitter";
-  inherit version;
-  inherit src;
+  inherit src version cargoSha256;
 
-  patches = [
-    # the web ui requires tree-sitter compiled to js and wasm
-    ./disable-web-ui.patch
-  ];
+  nativeBuildInputs = [ emscripten which ];
 
   postPatch = ''
     # needed for the tests
     rm -rf test/fixtures/grammars
     ln -s ${grammars} test/fixtures/grammars
   '';
 
+  # Compile web assembly with emscripten. The --debug flag prevents us from
+  # minifying the JavaScript; passing it allows us to side-step more Node
+  # JS dependencies for installation.
+  preBuild = ''
+    HOME=/tmp
+    bash ./script/build-wasm --debug
+  '';
+
+  # test result: FAILED. 120 passed; 13 failed; 0 ignored; 0 measured; 0 filtered out
+  doCheck = false;
```

Comment:
It's somewhat flaky (on master as well).  Every run has a different number of failures; here's my best run so far:

```
failures:
    tests::corpus_test::test_feature_corpus_files
    tests::corpus_test::test_real_language_corpus_files
    tests::parser_test::test_parsing_with_timeout_and_no_completion
    tests::query_test::test_query_captures_with_duplicates
    tests::query_test::test_query_matches_in_language_with_simple_aliases

test result: FAILED. 128 passed; 5 failed; 0 ignored; 0 measured; 0 filtered out
```